Library/Homebrew/extend/os/linux/software_spec.rb

```diff
@@ -1,6 +1,40 @@
 # typed: strict
 # frozen_string_literal: true
 
+class SoftwareSpec
+  extend T::Sig
+
+  GNU_GCC_REGEXP = T.let(/^(gcc)$|^(gcc@(4\.9|[5-9]|[10-99]))$/.freeze, Regexp)
```

Comment:
```suggestion
  GNU_GCC_REGEXP = T.let(/^(gcc)$|^(gcc@(4\.9|[5-9]|[10-99]))$/.freeze, Regexp).freeze 
```

or something might work (@SeekingMeaning can probably help)

Reply:
I tried that and I can't get both `brew typecheck` and `brew style` pass on this line.

Reply:
```suggestion
  GNU_GCC_REGEXP = T.let(/^(gcc)$|^(gcc@(4\.9|[5-9]|[10-99]))$/.freeze, Regexp) # rubocop:disable Style/MutableConstant
```

or:

```suggestion
  GNU_GCC_REGEXP = /^(gcc)$|^(gcc@(4\.9|[5-9]|[10-99]))$/.freeze
```

```patch
-# typed: strict
+# typed: true
```

Reply:
I think the latter `true` is a bit nicer, personally.

Reply:
Done